pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -220,19 +217,14 @@ func (m *managerImpl) synchronize(diskInfoProvider DiskInfoProvider, podFunc Act
 	glog.V(3).Infof("eviction manager: synchronize housekeeping")
 	// build the ranking functions (if not yet known)
 	// TODO: have a function in cadvisor that lets us know if global housekeeping has completed
-	if m.dedicatedImageFs == nil {
-		hasImageFs, ok := diskInfoProvider.HasDedicatedImageFs()
-		if ok != nil {
-			return nil
-		}
-		m.dedicatedImageFs = &hasImageFs
-		m.resourceToRankFunc = buildResourceToRankFunc(hasImageFs)
-		m.resourceToNodeReclaimFuncs = buildResourceToNodeReclaimFuncs(m.imageGC, m.containerGC, hasImageFs)
+	if len(m.resourceToRankFunc) == 0 || len(m.resourceToNodeReclaimFuncs) == 0 {
```

Comment:
I don't think you should change the code here. HasDedicatedImageFs is need to build for resourceToNodeReclaimFunc. Before my change, the eviction manager monitors both nodefs and imagefs usage and perform eviction when there is outofdisk condition. This part should not be changed.